### PR TITLE
Handle skip slots in fork choice

### LINKED ISSUE #770: [0.10.1]: Properly handle skip slots in fork choice
Spec PR: https://github.com/ethereum/eth2.0-specs/pull/1579

### PR DESCRIPTION
**Pull request by twoeths**
## Goal
+ Close #770

### CODE REVIEW

File: packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts [hunk 1]
```diff
@@ -391,11 +404,8 @@ export class StatefulDagLMDGHOST implements ILMDGHOST {
       return true;
```
Comment (twoeths) [inline packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts:404]: Not related to this PR, I think this condition should be the other way as the other conditions:
```
if (computeSlotsSinceEpochStart(this.config, getCurrentSlot(this.config, this.genesisTime)) >=
      this.config.params.SAFE_SLOTS_TO_UPDATE_JUSTIFIED)
      return false;
```
The reason is we want to check other conditions as well to make sure we have good justified checkpoints.
What do you think @wemeetagain ? Although it's the same to the spec, I think it's more reasonable to switch the condition, or make it the last condition in this function.

Comment (wemeetagain) [reply packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts:404]: :+1: that makes sense to me

File: packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts [hunk 2]
```diff
@@ -290,13 +292,24 @@ export class StatefulDagLMDGHOST implements ILMDGHOST {
       slot,
       blockRoot,
       stateRoot: stateRootBuf,
-      parent: this.nodes[parentRoot],
+      parent: this.nodes[parentRoot],
```
Comment (wemeetagain) [inline packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts]: unnecessary space

File: packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts (hunk 1 above)
Comment (twoeths) [reply packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts:404]: I see the spec and Prysmatic implement the same to this. Should we make a PR to the spec as well?

Comment (wemeetagain) [reply packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts:404]: We should definitely clarify what the intended behavior is

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**